File: problems/trademarket/solutions/trademarket.js
const getMaxProfit = input => {
    const prices = input.split(' ').map(price => parseInt(price));

    let maxProfit = prices[1] - prices[0];
    let currentMin = prices[0];

    for (let i = 1; i < prices.length; i++) {
        maxProfit = Math.max(maxProfit, prices[i] - currentMin);
        currentMin = Math.min(currentMin, prices[i]);
    }

    return maxProfit;
}



const chunks = [];
process.stdin.on('data', d => chunks.push(d));

process.stdin.on('end', () => {
    const data = chunks.join('').trim();
    console.log(getMaxProfit(data));
});

File: problems/trademarket/solutions/trademarket.py
def get_max_profit(prices):
    max_profit = prices[1] - prices[0]
    current_min = prices[0]

    for price in prices[1:]:
        max_profit = max(max_profit, price - current_min)
        current_min = min(current_min, price)

    return max_profit

prices = [int(x) for x in input().split()]

print(get_max_profit(prices))